spec/factories/course_forum_topics.rb

```diff
@@ -1,12 +1,20 @@
 # frozen_string_literal: true
 FactoryGirl.define do
   factory :forum_topic, class: Course::Forum::Topic.name do
-    forum
+    transient do
+      course { build(:course) }
```

Comment:
FactoryGirl doesnt care where the definition of the attribute is, so long the object responds to the appropriate methods. So keeping the factory there and inheriting it as a parent is the correct thing to do.